### PR TITLE
Credential Caching for the process-creds command

### PR DESCRIPTION
**Pull request by jakelerner-czi**
Very much a work in progress, creating PR now for visibility.

### CODE REVIEW

**Comment by kuannie1** [comment]
My question: can we move a lot of the logic outside the cmd package? Or explain (in the README.md) what the command is & how to use it.

**Comment by jakelerner-czi** [comment]
> My question: can we move a lot of the logic outside the cmd package? Or explain (in the README.md) what the command is & how to use it.

I agree! I think the plan is actually to move a lot of this logic outside of aws-oidc altogether (ie into go-misc). These changes to alter the command or how it's used at all, just provide transparent caching behind the scenes.

File: pkg/creds_process/processed-cred.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
+package cred
```
Comment (edulop91) [inline pkg/creds_process/processed-cred.go:1]: `package creds_process`

to match the dir name

File: pkg/creds_process/creds_cache.go [hunk 2]
```diff
@@ -0,0 +1,108 @@
+package cred
```
Comment (edulop91) [inline pkg/creds_process/creds_cache.go:1]: same here with naming this package creds_process to match the dir structure

**Review by edulop91 — APPROVED** [review]

**Pull request merged**